torch/distributed/fsdp/_flat_param.py

```diff
@@ -543,7 +543,7 @@ def __init__(
         self._device_handle = _FSDPDeviceHandle.from_device(self.device)
         self.process_group = process_group
         if self._use_fake_all_gather or self._use_fake_reduce:
-            self._fake_process_group = FakeProcessGroup(
+            self._fake_process_group = FakeProcessGroup._create_internal(
```

Comment:
I'm actually kind of skeptical about this use site, it sort of feels like potentially this is buggy LOL